### DISCUSSION TITLE
- vhost flag

### DISCUSSION THREAD

**Discussion by c4ir0** [projectdiscovery/httpx#780] [Q&A] [answered]
Hey 
I wonder what exactly the -vhost is,
###   IS it equal to running such command:

`$ ffuf -w subdomains-top1mil-20000.txt -H "Host: FUZZ.site.com" -u "http://site.com"`
why this command ? 
Some subdomains aren’t always hosted in publically accessible DNS results, such as development versions of a web application or administration portals. Instead, the DNS record could be kept on a private DNS server or recorded on the developer’s machines in their /etc/hosts file (or c:\windows\system32\drivers\etc\hosts file for Windows users) which maps domain names to IP addresses.

Because web servers can host multiple websites from one server when a website is requested from a client, the server knows which website the client wants from the Host header. We can utilize this host header by making changes to it and monitoring the response to see if we’ve discovered a new website.
[https://debprasadbanerjee502.medium.com/the-secret-trick-for-subdomain-enumeration-91b28be2b957](https://debprasadbanerjee502.medium.com/the-secret-trick-for-subdomain-enumeration-91b28be2b957) 

### so are they equivalent ? 
thx ❤

**Comment by olearycrew** [comment, marked as answer]
Hi @c4ir0 - that's not funtionality that we have planned for httpx itself, buit we have a project that does exactly that: https://github.com/projectdiscovery/alterx.

You also can then pipe those together, something like:

`alterx -list list.txt | httpx`
